src/adapter/src/optimize/mod.rs

```diff
@@ -151,7 +152,34 @@ where
 // Optimizer configuration
 // -----------------------
 
-// Feature flags for the optimizer.
+/// Feature flags for the optimizer.
+///
+/// To add a new feature flag, do the following steps:
+///
+/// 1. To make the flag available to all stages in our [`Optimize`] pipelines:
+///    1. Add the flag as an [`OptimizerConfig`] field.
+///
+/// 2. To allow users to set a system-wide override for this feature flag:
+///    1. Add the flag to the `feature_flags!(...)` macro call.
+///    2. Extend the `From<&SystemVars>` implementation for [`OptimizerConfig`].
+///
+/// 3. To enable `EXPLAIN ... WITH(...)` overrides:
+///    1. Add the flag as a [`mz_repr::explain::ExplainConfig`] field.
+///    2. Add the flag to the `impl_simple_options!(ExplainPlanOption {...})`
+///       macro call.
+///    3. Extend the `TryFrom<ExplainPlanOptionExtracted>` implementation for
+///       [`mz_repr::explain::ExplainConfig`].
+///    4. Extend the `OverrideFrom<ExplainContext>` implementation for
+///       [`OptimizerConfig`].
+///
+/// 4. To enable `CLUSTER ... FEATURES(...)` overrides:
+///    1. Add the flag to the `optimizer_feature_flags!(...)` macro call.
+///    2. Add the flag to the `impl_simple_options!(ClusterFeature {...})` macro
+///       call.
+///    2. Extend the `let optimizer_feature_overrides = ...` call in
+///       `plan_create_cluster`.
+///    4. Extend the `OverrideFrom<OptimizerFeatureOverrides>` implementation
+///       for [`OptimizerConfig`].
```

Comment:
You could also mention potentially adding
```
self.reoptimize_imported_views = true;
```
to `impl OverrideFrom<OptimizerFeatureOverrides> for OptimizerConfig` and to `impl OverrideFrom<ExplainContext> for OptimizerConfig`.

Reply:
Discussed offline. I've changed the commit to promote `reoptimize_imported_views` to a feature flag.